### DISCUSSION TITLE
Image optimization URL doesn't work when images introduced to file system after nextJS server started

### DISCUSSION THREAD

**Discussion by simontong** [vercel/next.js#32952] [Ideas]
### What version of Next.js are you using?

12.0.7

### What version of Node.js are you using?

16.3.0

### What browser are you using?

all

### What operating system are you using?

linux

### How are you deploying your application?

yarn dev / yarn build && yarn start

### Describe the Bug

NextJS response with optimized images that existed on the file system BEFORE I started the server. If I upload an image and place it in the `public/uploads/foobar.jpg` path when I attempt to display that image on the page it just hangs and doesn't render anything. If I restart the server and access the URL the image is rendered as expected.

I noticed when I try to access an image that doesn't exist the connection hangs indefinitely.

I also noticed that when if I'm running the build version (not dev) of nextjs and I introduce a file after the server is running and attempt to access it I get this error in the console:

```
TypeError: res.hasHeader is not a function
    at setRevalidateHeaders (/path/to/app/node_modules/next/dist/server/send-payload.js:18:37)
    at Object.sendRenderResult (/path/to/app/node_modules/next/dist/server/send-payload.js:51:9)
```

### Expected Behavior

Expect images to be rendered that get uploaded after the next server is already running. Also when an image is missing to respond with 404Expect images to be rendered that get uploaded after the next server is already running. Also when an image is missing to respond with 404

### To Reproduce

1. start next server
2. Upload image into public/uploads/abc.jpg
3. access an image optimize URL. eg. http://localhost:3000/_next/image?url=/uploads/abc.jpg&w=640&q=75
4. restart the server and the image will render as expected

**Comment by simontong** [comment]
Update:

When I tried using node version v15.14.0 it still doesn't render the image until I reload the server but now instead of it hanging it will say "The requested resource isn't a valid image." when the image 100% exists in the location I'm trying to access

**Comment by balazsorban44** [comment]
I don't believe it is an actual bug, but more of a feature request, since you are describing some functionality that just does not exist as of today. (Unless we document it otherwise. Can you confirm?).

From the source code, it looks like we only look for changes in the `pages` directory in dev mode: https://github.com/vercel/next.js/blob/a5bfc1eecc34e176f9f27955fde7b16e17de3c5e/packages/next/server/dev/next-dev-server.ts#L254

If you actually mean adding files when running in production (eg `next start`), that's expected behavior. See the docs:

https://nextjs.org/docs/basic-features/static-file-serving

>  Only assets that are in the `public` directory at [build time](https://nextjs.org/docs/api-reference/cli#build) will be served by Next.js. Files added at runtime won't be available. We recommend using a third-party service like [AWS S3](https://aws.amazon.com/s3/) for persistent file storage.

**Comment by simontong** [comment]
The problem is that I place a file in `./public/uploads/myimage.jpg` while the production server is running and when I access it via `https://website.com/_next/image?url=/uploads/myimage.jpg` it says "The requested resource isn't a valid image.". 

If I restart the server (in production) then and refresh the same URL then the image appears as expected.

Basically the way it is right now I would need to restart the server every time someone uploads an image in order to display said image using the <Image/> nextjs component

**Reply by balazsorban44** [reply]
See the docs link from my previous comment.
